src/v/cloud_storage/remote_segment_index.cc

```diff
@@ -363,6 +367,25 @@ void remote_segment_index_builder::consume_batch_start(
         }
     }
     _window += size_on_disk;
+
+    // Update stats
+    if (_stats.has_value()) {
+        if (is_config) {
+            _stats->get().total_conf_records += delta;
+        } else {
+            _stats->get().total_data_records += delta;
+        }
+        if (_stats->get().base_rp_offset == model::offset{}) {
+            _stats->get().base_rp_offset = hdr.base_offset;
+        }
+        _stats->get().last_rp_offset = std::max(
+          _stats->get().last_rp_offset, hdr.last_offset());
```

Comment:
nit: should this field always be updated to `hdr.last_offset()` as it will always be higher than `last_rp_offset`?